spotbugs/src/main/java/edu/umd/cs/findbugs/ba/bcp/PatternMatcher.java

```diff
@@ -448,11 +450,11 @@ private MatchResult matchLocation(Location location) throws DataflowAnalysisExce
     }
 
     private void debug(String s) {
-        if (!DEBUG) {
+        if (!LOG.isDebugEnabled()) {
             throw new IllegalStateException("Only call if DEBUG is true");
         }
-        System.out.print("                                            ".substring(0, depth));
-        System.out.println(s);
+        String indent = "                                            ".substring(0, depth);
```

Comment:
This really should be replaced by `StringUtils.repeat` or similar functionality.